include/evmc.h

```diff
@@ -140,48 +140,97 @@ typedef void (*evmc_get_block_hash_fn)(struct evmc_uint256be* result,
 
-/// The execution status code.
-enum evmc_status_code {
-    EVMC_SUCCESS = 0,               ///< Execution finished with success.
-    EVMC_FAILURE = 1,               ///< Generic execution failure.
-    EVMC_OUT_OF_GAS = 2,
-    EVMC_UNDEFINED_INSTRUCTION = 3, ///< Unknown instruction encountered by the VM.
-    EVMC_BAD_JUMP_DESTINATION = 4,
-    EVMC_STACK_OVERFLOW = 5,
-    EVMC_STACK_UNDERFLOW = 6,
-    EVMC_REVERT = 7,                ///< Execution terminated with REVERT opcode.
-
-    /// Tried to execute an operation which is restricted in static mode.
-    EVMC_STATIC_MODE_VIOLATION = 8,
-
-    /// The dedicated INVALID instruction was hit.
-    EVMC_INVALID_INSTRUCTION = 9,
-
-    /// Tried to read outside memory bounds.
-    ///
-    /// An example is RETURNDATACOPY reading past the available buffer.
+/**
+ * The execution status code.
+ *
+ * Successful execution is represented by ::EVMC_SUCCESS having value 0.
+ * Positive values represent failures defined by VM specifications with generic
+ * ::EVMC_FAILURE code of value 1.
+ * Negative values represent VM internal errors and might be not recoverable.
+ * The generic representant of errors is ::EVMC_INTERNAL_ERROR.
+ * An EVM implementation MAY return negative status codes that are not specified
+ * in EVMC.
+ */
+enum evmc_status_code
+{
+    /** Execution finished with success. */
+    EVMC_SUCCESS = 0,
+
+    /** Generic execution failure. */
+    EVMC_FAILURE = 1,
+
+    /**
+     * Execution terminated with REVERT opcode.
+     *
+     * In this case the amount of gas left MAY be non-zero and additional output
+     * data MAY be provided in ::evmc_result.
+     */
+    EVMC_REVERT = 2,
+
+
+    /** The execution has run out of gas. */
+    EVMC_OUT_OF_GAS = 3,
+
+
+    /**
+     * The execution has attempted to put more items on the EVM stack
+     * than the specified limit.
+     */
+    EVMC_STACK_OVERFLOW = 4,
+
+
+    /** Execution of an opcode has required more items on the EVM stack. */
+    EVMC_STACK_UNDERFLOW = 5,
+
+    /** Execution has violated the jump destination restrictions. */
+    EVMC_BAD_JUMP_DESTINATION = 6,
+
+    /** An undefined instruction has been encountered. */
+    EVMC_UNDEFINED_INSTRUCTION = 7,
+
+    /**
+     * The designated INVALID instruction has been hit during execution.
+     *
+     * The EIP-141 (https://github.com/ethereum/EIPs/blob/master/EIPS/eip-141.md)
+     * defines the instruction 0xfe as INVALID instruction to indicate execution
+     * abortion coming from high-level languages. This status code is reported
+     * in case this INVALID instruction has been encountered.
+     */
+    EVMC_INVALID_INSTRUCTION = 8,
+
+    /** Tried to execute an operation which is restricted in static mode. */
+    EVMC_STATIC_MODE_VIOLATION = 9,
+
+    /**
+     * Tried to read outside memory bounds.
+     *
+     * An example is RETURNDATACOPY reading past the available buffer.
+     */
     EVMC_INVALID_MEMORY_ACCESS = 10,
```

Comment:
I think this should come before `static mode`.

Reply:
Why?

Reply:
I think a logical order would be:
- invalid/unknown instructions
- invalid number of arguments for instructions
- invalid arguments (bad jump, invalid memory access)
- invalid state (static mode violation)

Reply:
No comment on this nice ordered list? 😉

Reply:
I did it this way...

Reply:
I don't think so :)

Reply:
```
EVMC_INVALID_MEMORY_ACCESS = 9,
EVMC_CALL_DEPTH_EXCEDED = 10,
EVMC_STATIC_MODE_VIOLATION = 11,
```

What am I missing?

Reply:
This part:
- invalid/unknown instructions
- invalid number of arguments for instructions
- invalid arguments (bad jump)

The order now is argument count, bad jump, invalid/unknown, etc.

Reply:
Reordered again then.